kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,242 @@
+        """
+        # TODO: When can we recover? What are the failures? How do we recover?
+        if not failures:
+            return
+        for failure in failures:
+            self.log.error(failure)
+        raise exceptions.UnrecoverableActorFailure()
+
+
+class RunTask(ECSBaseActor):
+
+    """Register and run a task on ECS.
+
+    This actor will loop indefinitely until the task is complete.
+
+    **Options**
+
+    :task_definition:
+      String of path to the Task Definition file template
+      Must be a local file path
+      Parameters to be interpolated must be of the form %VAR%
+
+    :parameters:
+      A dictionary of key/value pairs used to fill in the parameters for the
+      Task Definition template
+
+    :region:
+      AWS region (or zone) name, such as us-west-2 or eu-west-1
+
+    :cluster:
+      ECS cluster in the region to launch the task into
+
+    :count:
+      How many tasks to run
+
+    :skip_wait:
+      Whether to skip waiting for the tasks to complete
+
+    **Examples**
+
+    .. code-block:: yaml
+
+       actor: aws.ecs.RunTask
+       desc: Run migrations
+       options:
+          task_definition: nextdoor-migrate.json
+          region: us-west-2
+          cluster: us1-internal
+
+    **Dry Mode**
+
+    Will only attempt to interpolate env vars into the task definition.
+    """
+
+    all_options = {
+        'task_definition':
+            (str, REQUIRED,
+             'String of path to the Task Definition file template. '
+             'Must be a local file path. '
+             'Parameters to be interpolated must be of the form %VAR%'),
+        'parameters': (dict, {},
+                       'A dictionary of key/value pairs '
+                       'used to fill in the parameters for the template'),
+        'region': (str, REQUIRED,
+                   'AWS region (or zone) name, like us-west-2 or eu-west-1'),
+        'cluster': (str, REQUIRED,
+                    'ECS cluster in the region to launch the task into'),
+        'count': (int, 1, 'How many tasks to run'),
+        'skip_wait': (bool, False,
+                      'Whether to skip waiting for the tasks to complete')
+    }
+
+    @gen.coroutine
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision').
+        """
+        if self._dry:
+            self.log.warning('Would register task')
+            return task_definition['family'], '1'
+
+        response = self.ecs_conn.register_task_definition(**task_definition)
+
+        # Parse data from the server's response.
+        task_definition = response['taskDefinition']
+        family = task_definition['family']
+        revision = task_definition['revision']
+
+        self.log.info('Task definition {0}:{1} registered'.format(
+            family, revision))
+        return family, revision
```

Comment:
Any method wrapped in `gen.coroutine()` cannot `return ...` a value. Instead, you raise it. See this example:

```
@gen.coroutine
def hello_world(self):
    ret = yield self.client.get_hello_world_message()
    raise gen.Return(ret)
```